src/Consolonia.Core/Drawing/RenderTarget.cs

```diff
@@ -180,7 +183,9 @@ private void RenderToDevice()
                     _cache[x++, y] = Pixel.Empty;
```

Comment:
This X coordinate calculation is now out of sync with WritePixel. Also, In WritePixel we should advance y + 1 in case end of the line hit. Because if this and that calculations are out of sync we do unecessary Caret moves.